pandas/indexes/multi.py

```diff
@@ -71,8 +71,10 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
                 name=None, **kwargs):
 
         # compat with Index
-        if name is not None:
```

Comment:
``names`` AND ``name`` need to be correctly handled; look at ``_shallow_copy`` for this logic (which may need to be moved to a method)